File: Easy/101. Symmetric Tree/ans.js
var isSymmetric = function(root) {
    
    var roots=[root.left,root.right];
    var l,r;
    while(roots.length>0){
        l=roots.shift();
        r=roots.shift();
        if(!l && !r) continue;
        if(!l || !r) return false;
        if(l.val!==r.val) return false;
        roots.push(l.left,r.right);
        roots.push(l.right,r.left);
    }
    
    return true;
    
    
};

File: Easy/104. Maximum Depth of Binary Tree/ans.py
class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if not root:
            return 0
        stack,depth = [(root,1)],0
        while stack:
            cur,cur_depth = stack.pop()
            if cur:
                depth = max(depth,cur_depth)
                if cur.left:
                    stack.append((cur.left,cur_depth+1))
                if cur.right:
                    stack.append((cur.right,cur_depth+1))
        return depth